### ISSUE TITLE
Documentation improvement

### ISSUE THREAD

**Issue by Asu4ni** [gnocchixyz/gnocchi#340]
I'm feeling that our doc is lacking tons of details and there are so much stuf that one can't figure out until browsing through the codes. I'll give some examples down below, and I'd help adding the content now and then. 

the list of things to be improved:
- [ ] Glossary: needs more items such as timespan, archive policy rule, resource type, etc. and anchors should be placed for linking.
- [ ] Details of each entities: for entities like resource, metric, measure, etc., the relation between them (e.g. resource contains metrics), the attributes of them (e.g. metric has creator) should be explicitly explained, listed and illustrated in a page.
- [ ] Aggregation method: a complete list showing all ones and the explanations of them.
- [ ] REST API: the usage of each api should be explained in details such as what attribute names are invalid.

If I come up more or someone suggests some, I'll update the list

**Comment by Asu4ni** [comment]
@jd one question. How can I preview the result of my alterations?

**Comment by jd** [comment]
@Asu4ni Run `tox -e docs-gnocchi.xyz` and check `doc/build/html` for output.

**Comment by Asu4ni** [comment]
I cloned the repo and ran the tox command. The page was correctly displayed.
Afterwards, I modified the rst file and ran the command. The html files were updated (checked the modified time) but the contents were the same. Even if I commited my change, it still didn't work.
I'm digging into the doc of sphinx now, but if someone catches what I've missed, please tell me. Thanks!

**Comment by Asu4ni** [comment]
The error message of the above problem which might be the cause.
`Exception: Versioning for this project requires either an sdist tarball, or access to an upstream git repository. It's also possible that there is a mismatch between the package name in setup.cfg and the argument given to pbr.version.VersionInfo. Project name gnocchi was given, but was not able to be found.`

**Comment by jd** [comment]
@Asu4ni ah don't bother, it's likely sphinx versioncontrib give you problems. Just run `tox -e docs` in your case, that'll be enough.

**Comment by Asu4ni** [comment]
I think we need to distinguish between 'incoming measures' and 'processed measures'.

I'm considering this because our [glossary](http://gnocchi.xyz/glossary.html#) in the document have the item 'measure'. But it does not explicitly show which one it represents(incoming or processed).

I'd like to gain some thoughts from others' opinion. Mine is that we should use two words for these two concepts. Maybe 'measure' for incoming and 'aggregates' for processed.

Or I simply mistook the concepts. XD

**Comment by jd** [comment]
You're right @Asu4ni, I think it's a good idea to distinguish both and your terms are good. Go ahead if you want to add those and update the glossary and doc!
